packages/framework/src/client/scheduler.js

```diff
@@ -1,20 +1,31 @@
 import debugFactory from 'debug';
-import { find } from 'lodash';
+import { find, groupBy, isArray, isEmpty, isEqual, isNil, uniq } from 'lodash';
 import ResourceRequest, { STATUS } from './resource-request';
 
 const debug = debugFactory( 'fresh-data:scheduler' );
 
 export default class Scheduler {
+	/**
+	 * Creates a new Scheduler
+	 * @param {Object} operations The mapping of operation names to functions
+	 * @param {function} setTimeout The function to set a time (defaults to window.setTimeout)
+	 * @param {function} clearTimeout The function to clear a timer by id (defaults to window.setTimeout)
+	 */
 	constructor(
-		fetch,
+		operations,
 		setTimeout = window.setTimeout,
 		clearTimeout = window.clearTimeout,
 	) {
-		this.fetch = fetch;
+		this.debug = debugFactory( 'fresh-data:scheduler' );
+
+		this.operations = operations;
 		this.requests = [];
 		this.setTimeout = setTimeout;
 		this.clearTimeout = clearTimeout;
 		this.timeoutId = null;
+
+		this.dataRequested = () => this.debug( 'warning: dataRequested called before setDataHandlers' );
+		this.dataReceived = () => this.debug( 'warning: dataReceived called before setDataHandlers' );
```

Comment:
This is a good point. In the previous implementation it was necessary because of the integration of redux and a provider where `dataRequested` and `dataReceived` were dispatch-wrapped properties from redux.

This may no longer be the case. I'll review the providers from `react-provider` and the one in `wc-api` to make sure.

Reply:
Okay, I've taken a look and I think this is a good thing to be done. However, given that this PR is over 800 lines long and in a long chain of other PRs, I'd like to propose a future PR for this--especially given that this will be a breaking change for `fresh-data`. I've created issue #210 to track this.